### PR TITLE
SOL-13239:Updated Go version and Monstache version.

### PR DESCRIPTION
**Pull request by Eknath-Sable**

### CODE REVIEW

**Review by ratneshdubeysolidatus — APPROVED** [review]
LGTM

File: Solidatus-README.md [hunk 1]
```diff
@@ -24,3 +24,7 @@ Golang has 3 vulnerabilities CVE-2023-29409, CVE-2023-29406, CVE-2023-39533 whic
 # 2023-10-18
 
 Golang has 1 vulnerability CVE-2023-39320 which was fixed in 1.21.3 hence manually updated go.mod to use 1.21.3. (latest on main branch was 1.21)
+
+# 2024-01-10
+
+Golang has 2 vulnerability CVE-2023-39326 and CVE-2023-45285 which was fixed in 1.21.6 hence manually updated go.mod to use 1.21.6. (latest on main branch was 1.21)
```
Comment (mix4242) [inline Solidatus-README.md]: Should be
```suggestion
Golang has 2 vulnerability CVE-2023-39326 and CVE-2023-45285 which was fixed in 1.21.6 hence manually updated go.mod to use 1.21.6. (latest on main branch was 1.21.3)
```
right?

File: monstache.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ var chunksRegex = regexp.MustCompile("\\.chunks$")
 var systemsRegex = regexp.MustCompile("system\\..+$")
 var exitStatus = 0
 
-const version = "6.7.10.3"
+const version = "6.7.17"
```
Comment (mix4242) [inline monstache.go]: Why have we bumped version beyond `6.7.10.x`?

**Review by mix4242 — CHANGES_REQUESTED** [review]

File: monstache.go (hunk 2 above)
Comment (Eknath-Sable) [reply monstache.go]: There have been several vulnerabilities reported for Go libraries that are picked up in our internal vuln scans. While these do not pose an attack vector for our use case. We currently use Monstache 6.7.10 while the latest version is 6.7.17.This is implemented as per SOL-13251

Comment (mix4242) [reply monstache.go]: Right, but you're not bringing in any changes from the latest version 6.7.17 here. You're just bumping the Go version.

**Review by mix4242 — CHANGES_REQUESTED** [review]

File: monstache.go (hunk 2 above)
Comment (Eknath-Sable) [reply monstache.go]: we have bumped the Go version from 1.21.3 to 1.21.6 for CVE's and We upgraded the monstache version to 6.7.17 without bringing any new feature.

Comment (mix4242) [reply monstache.go]: Won't it be confusing, bumping this version of monstache to `6.7.17`, when it actually has the monstache code base equivalent of `6.7.10`? Also how will one distinguish this `6.7.17` from the official (but very different) `6.7.17`?

**Review by mix4242 — CHANGES_REQUESTED** [review]

File: monstache.go (hunk 2 above)
Comment (ratneshdubeysolidatus) [reply monstache.go]: Yes, it's confusing bumping the version without actual code changes. Shall we document this clearly in the code base that this version is based on 6.7.10 code. Or welcome your suggestion for the other versioning approach.

Comment (mix4242) [reply monstache.go]: I think the best is to carry on with the versioning approach that we had - keep the full monstache version & bump the extra number at the end, treating it like a build number :)

Comment (Eknath-Sable) [reply monstache.go]: @mix4242 .. Thanks for your inputs . What version you are suggesting here ? 6.7.17.X?

Comment (mix4242) [reply monstache.go]: Keep the version we had, just bump the last number :)

**Review by mix4242 — APPROVED** [review]

**Review by rrughani — APPROVED** [review]

File: monstache.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ var chunksRegex = regexp.MustCompile("\\.chunks$")
 var systemsRegex = regexp.MustCompile("system\\..+$")
 var exitStatus = 0
 
-const version = "6.7.17"
+const version = "6.7.10+4"
```
Comment (ratneshdubeysolidatus) [inline monstache.go:85]: I am just curious about this; do we really have a version naming convention like 6.7.10+4?

**Pull request merged**